src/robot.coffee

```diff
@@ -247,6 +264,28 @@ class Robot
   # Returns nothing.
   # Returns before executing callback
   receive: (message, cb) ->
+    # When everything is finished (down the middleware stack and back up),
+    # pass control back to the robot
+    receiveMiddlewareDone = () ->
+      if cb?
+        process.nextTick -> cb true
+
+    @middleware.receive.execute(
+      {response: new Response(this, message)}
```

Comment:
My current idea for response middleware is to have it just be a piece of receive middleware instead of an actual change to the core. Similar to [how gzipping works in Express](https://github.com/expressjs/compression/blob/master/index.js#L60-L108), we would just monkey patch the response object to send responses through some proxy code. That would give us room to start playing with ideas without requiring core modifications or finalized APIs (i.e. faster iteration).

Another use case for receive middleware is automatically doing LDAP group lookups and attaching LDAP groups to something listener middleware and the listener callback can then utilize. While you can get the `message` object out and attach to that (because it passes through), that is strange and non-intuitive. As a consumer of the framework, I would expect either my entire object passes through or it doesn't. Having an identical object type (Response) that doesn't actually pass down is pretty weird.